rpc/lib/server/handlers.go

```diff
@@ -127,10 +129,13 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger log.Logger) http.Han
 			WriteRPCResponseHTTP(w, types.RPCMethodNotFoundError(request.ID))
 			return
 		}
-		args, err := jsonParamsToArgsRPC(rpcFunc, request.Params)
-		if err != nil {
-			WriteRPCResponseHTTP(w, types.RPCInvalidParamsError(request.ID, errors.Wrap(err, "Error converting json params to arguments")))
-			return
+		var args []reflect.Value
+		if len(request.Params) > 0 {
```

Comment:
as far as I can tell, now that params is not a pointer, we shouldn't need this guard, but it's fine to leave it

Reply:
no, we need it because overwise this results in the following error: "params should be a map or array"